src/main/java/uk/nhs/ctp/service/factory/ReferencingContextFactory.java

```diff
@@ -1,18 +0,0 @@
-package uk.nhs.ctp.service.factory;
-
-import lombok.AllArgsConstructor;
-import org.springframework.stereotype.Component;
-import uk.nhs.ctp.entities.CdssSupplier;
-import uk.nhs.ctp.service.CdssSupplierService;
-import uk.nhs.ctp.service.fhir.ReferencingContext;
-
-@Component
-@AllArgsConstructor
-public class ReferencingContextFactory {
-
-  private CdssSupplierService cdssSupplierService;
-
-  public ReferencingContext load(CdssSupplier cdss) {
-    return new ReferencingContext(cdss.getReferencingType());
```

Comment:
Are these still required for implementing the referencing toggles? I feel like they may be unused now because we updated everything to use references, but implementing the toggle might still use them in future?
@john-herciu-bjss

Reply:
It still seems like a hugely complicated way of toggling resources. I think we can design something simpler (e.g. two implementations of an parameter-creation interface)

Reply:
They should be fine to remove, the plan was to remove a lot of these as part of the new referencing toggles work anyway.